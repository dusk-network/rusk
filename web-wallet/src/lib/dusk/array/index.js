export { default as arraysEqual } from "./arraysEqual";
export { default as arrayMax } from "./arrayMax";
export { default as arrayMaxByKey } from "./arrayMaxByKey";
export { default as shuffleArray } from "./shuffleArray";
export { default as findFirstNMatches } from "./findFirstNMatches";
