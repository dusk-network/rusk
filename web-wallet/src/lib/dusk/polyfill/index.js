/** @overview Import the module to load all polyfills at once */

import "./asyncIterator";
import "./promiseWithResolvers";

// eslint-disable-next-line import/no-unresolved
import "web-streams-polyfill/polyfill";
