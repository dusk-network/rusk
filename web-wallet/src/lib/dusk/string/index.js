export { default as calculateAdaptiveCharCount } from "./calculateAdaptiveCharCount";
export { default as cleanNumberString } from "./cleanNumberString";
export { default as hexStringToBytes } from "./hexStringToBytes";
export { default as isValidEvmAddress } from "./isValidEvmAddress";
export { default as makeClassName } from "./makeClassName";
export { default as middleEllipsis } from "./middleEllipsis";
export { default as randomUUID } from "./randomUUID";
export { default as truncateDecimals } from "./truncateDecimals";
