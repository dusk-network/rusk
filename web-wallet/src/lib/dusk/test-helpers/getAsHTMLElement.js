/**
 * Utility function to force the type checker to see
 * the return value of `querySelector` as a HTMLElement.
 *
 * @param {HTMLElement} container
 * @param {String} selector
 * @returns {HTMLElement}
 */
const getAsHTMLElement = (container, selector) =>
  /** @type {HTMLElement} */ (container.querySelector(selector));

export default getAsHTMLElement;
