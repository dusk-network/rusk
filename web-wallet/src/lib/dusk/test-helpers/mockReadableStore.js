import { get, writable } from "svelte/store";

/**
 * @template T
 * @param {T} initialValue
 */
function mockReadableStore(initialValue) {
  const store = writable(initialValue);
  const { set, subscribe } = store;
  const getMockedStoreValue = () => get(store);

  /** @param {T} value */
  const setMockedStoreValue = (value) => set(value);

  return {
    getMockedStoreValue,
    setMockedStoreValue,
    subscribe,
  };
}

export default mockReadableStore;
