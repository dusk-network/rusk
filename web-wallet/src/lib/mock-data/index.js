export { default as addresses } from "./addresses";
export { default as cacheHistory } from "./cache-history";
export { default as cacheSpentNotes } from "./cache-spent-notes";
export { default as cacheUnspentNotes } from "./cache-unspent-notes";
export { default as stakeInfo } from "./stakeInfo";
export { default as transactions } from "./transactions";
