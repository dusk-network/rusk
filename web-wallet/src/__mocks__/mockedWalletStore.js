import { ProfileGenerator } from "$lib/vendor/w3sper.js/src/profile";

import { stakeInfo } from "$lib/mock-data";

import { mockReadableStore } from "$lib/dusk/test-helpers";

const seed = new Uint8Array(64);
const seeder = () => seed;
const profileGenerator = new ProfileGenerator(seeder);
const profiles = [
  await profileGenerator.default,
  await profileGenerator.next(),
  await profileGenerator.next(),
];
const currentProfile = profiles[0];
const shieldedBalance = {
  spendable: 50_000_000_000_000n,
  value: 2_345_000_000_000n,
};
const publicBalance = { nonce: 1234n, value: shieldedBalance.value / 2n };

/** @type {WalletStoreContent} */
const content = {
  balance: { publicBalance, shieldedBalance },
  currentProfile,
  initialized: true,
  minimumStake: 1_000_000_000_000n,
  profiles,
  stakeInfo,
  syncStatus: {
    error: null,
    from: 0n,
    isInProgress: false,
    last: 0n,
    progress: 0,
  },
};

const mockedWalletStore = mockReadableStore(content);

export default mockedWalletStore;
