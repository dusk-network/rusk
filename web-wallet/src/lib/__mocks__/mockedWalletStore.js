import { ProfileGenerator } from "$lib/vendor/w3sper.js/src/profile";

import { stakeInfo } from "$lib/mock-data";

import { mockReadableStore } from "$lib/dusk/test-helpers";

const seed = new Uint8Array(64);
const seeder = () => seed;
const profileGenerator = new ProfileGenerator(seeder);
const profiles = [
  await profileGenerator.default,
  await profileGenerator.next(),
  await profileGenerator.next(),
];

const currentProfile = profiles[0];
const shielded = { spendable: 50_000_000_000_000n, value: 2_345_000_000_000n };
const unshielded = { nonce: 1234n, value: shielded.value / 2n };

/** @type {WalletStoreContent & { currentProfile: NonNullable<WalletStoreContent["currentProfile"]> }} */
const content = {
  balance: { shielded, unshielded },
  currentProfile,
  initialized: true,
  minimumStake: 1_000_000_000_000n,
  profiles,
  stakeInfo,
  syncStatus: {
    error: null,
    from: 0n,
    isInProgress: false,
    last: 0n,
    progress: 0,
  },
};

const mockedWalletStore = mockReadableStore(content);

export default mockedWalletStore;
