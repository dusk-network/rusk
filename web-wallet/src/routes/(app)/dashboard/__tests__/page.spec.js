import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
import { cleanup } from "@testing-library/svelte";
import { renderWithSimpleContent } from "$lib/dusk/test-helpers";
import mockedWalletStore from "$lib/__mocks__/mockedWalletStore";
import { stakeInfo, transactions } from "$lib/mock-data";

import Dashboard from "../+page.svelte";
import { walletStore } from "$lib/stores";

vi.mock("$lib/stores", async (importOriginal) => {
  /** @type {WalletStore} */
  const original = await importOriginal();

  return {
    ...original,
    walletStore: {
      ...mockedWalletStore,
      getStakeInfo: () => Promise.resolve(stakeInfo),
      getTransactionsHistory: () => Promise.resolve(transactions),
    },
  };
});

vi.useFakeTimers();

describe("Dashboard", () => {
  afterEach(cleanup);

  afterAll(() => {
    vi.doUnmock("$lib/stores");
  });

  it("should render the dashboard page with the transactions after they are loaded", async () => {
    const { container } = renderWithSimpleContent(Dashboard, {});

    await vi.advanceTimersToNextTimerAsync();

    expect(container).toMatchSnapshot();
  });

  it("should render a card when there is an error getting transactions", async () => {
    const someError = new Error("some error message");
    const walletSpy = vi
      .spyOn(walletStore, "getTransactionsHistory")
      .mockRejectedValue(someError);
    const { container } = renderWithSimpleContent(Dashboard, {});

    await vi.advanceTimersToNextTimerAsync();

    expect(container).toMatchSnapshot();

    walletSpy.mockRestore();
  });
});
