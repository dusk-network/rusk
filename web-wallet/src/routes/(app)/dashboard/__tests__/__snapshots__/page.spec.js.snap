// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dashboard > should render a card when there is an error getting transactions 1`] = `
<div>
  <div
    class="dusk-card dusk-card--gap-default"
  >
     
    <div
      class="dusk-card__body-container"
    >
      <nav
        aria-label="Transaction and operations navigation"
        class="dashboard-nav"
      >
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="send"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Send
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M5,17.59L15.59,7H9V5H19V15H17V8.41L6.41,19L5,17.59Z"
              />
            </svg>
            
          </span>
           
        </a>
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="receive"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Receive
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,6.41L17.59,5L7,15.59V9H5V19H15V17H8.41L19,6.41Z"
              />
            </svg>
            
          </span>
           
        </a>
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="stake"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Stake
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 3C7.58 3 4 4.79 4 7V17C4 19.21 7.59 21 12 21S20 19.21 20 17V7C20 4.79 16.42 3 12 3M18 17C18 17.5 15.87 19 12 19S6 17.5 6 17V14.77C7.61 15.55 9.72 16 12 16S16.39 15.55 18 14.77V17M18 12.45C16.7 13.4 14.42 14 12 14C9.58 14 7.3 13.4 6 12.45V9.64C7.47 10.47 9.61 11 12 11C14.39 11 16.53 10.47 18 9.64V12.45M12 9C8.13 9 6 7.5 6 7S8.13 5 12 5C15.87 5 18 6.5 18 7S15.87 9 12 9Z"
              />
            </svg>
            
          </span>
           
        </a>
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="allocate"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Allocate
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,18A6,6 0 0,1 6,12C6,11 6.25,10.03 6.7,9.2L5.24,7.74C4.46,8.97 4,10.43 4,12A8,8 0 0,0 12,20V23L16,19L12,15M12,4V1L8,5L12,9V6A6,6 0 0,1 18,12C18,13 17.75,13.97 17.3,14.8L18.76,16.26C19.54,15.03 20,13.57 20,12A8,8 0 0,0 12,4Z"
              />
            </svg>
            
          </span>
           
        </a>
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="bridge"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Bridge
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M21,9L17,5V8H10V10H17V13M7,11L3,15L7,19V16H14V14H7V11Z"
              />
            </svg>
            
          </span>
           
        </a>
      </nav>
    </div>
     
  </div>
  
   
  <span>
    some text
  </span>
   
  <article
    class="transactions svelte-mwi4ip"
  >
    <header
      class="transactions__header svelte-mwi4ip"
    >
      <h3
        class="h4"
      >
        Transactions
      </h3>
       
      <a
        aria-disabled="false"
        class="dusk-anchor dusk-anchor--on-surface dusk-anchor-button dusk-anchor-button--variant--primary dusk-anchor-button--size--default transactions__footer-button"
        href="/some-base-path/dashboard/transactions"
      >
        <span
          class="dusk-anchor-button__text"
        >
          All transactions
        </span>
        
      </a>
    </header>
     
    <div
      class="dusk-suspense transactions-list__container"
    >
      <details
        class="dusk-error-details dusk-error-details__details dusk-suspense__error"
      >
        <summary
          class="dusk-error-details__summary"
        >
          Error getting transactions
        </summary>
         
        <pre
          class="dusk-error-details__error"
        >
          			
          some error message
          
		
        </pre>
      </details>
      
       
       
    </div>
  </article>
  
  
</div>
`;

exports[`Dashboard > should render the dashboard page with the transactions after they are loaded 1`] = `
<div>
  <div
    class="dusk-card dusk-card--gap-default"
  >
     
    <div
      class="dusk-card__body-container"
    >
      <nav
        aria-label="Transaction and operations navigation"
        class="dashboard-nav"
      >
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="send"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Send
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M5,17.59L15.59,7H9V5H19V15H17V8.41L6.41,19L5,17.59Z"
              />
            </svg>
            
          </span>
           
        </a>
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="receive"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Receive
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,6.41L17.59,5L7,15.59V9H5V19H15V17H8.41L19,6.41Z"
              />
            </svg>
            
          </span>
           
        </a>
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="stake"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Stake
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 3C7.58 3 4 4.79 4 7V17C4 19.21 7.59 21 12 21S20 19.21 20 17V7C20 4.79 16.42 3 12 3M18 17C18 17.5 15.87 19 12 19S6 17.5 6 17V14.77C7.61 15.55 9.72 16 12 16S16.39 15.55 18 14.77V17M18 12.45C16.7 13.4 14.42 14 12 14C9.58 14 7.3 13.4 6 12.45V9.64C7.47 10.47 9.61 11 12 11C14.39 11 16.53 10.47 18 9.64V12.45M12 9C8.13 9 6 7.5 6 7S8.13 5 12 5C15.87 5 18 6.5 18 7S15.87 9 12 9Z"
              />
            </svg>
            
          </span>
           
        </a>
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="allocate"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Allocate
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,18A6,6 0 0,1 6,12C6,11 6.25,10.03 6.7,9.2L5.24,7.74C4.46,8.97 4,10.43 4,12A8,8 0 0,0 12,20V23L16,19L12,15M12,4V1L8,5L12,9V6A6,6 0 0,1 18,12C18,13 17.75,13.97 17.3,14.8L18.76,16.26C19.54,15.03 20,13.57 20,12A8,8 0 0,0 12,4Z"
              />
            </svg>
            
          </span>
           
        </a>
        
        <a
          class="dusk-anchor dusk-anchor--on-surface dashboard-nav__item"
          href="bridge"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="dashboard-nav__item-label svelte-1opr8zc"
          >
            Bridge
          </span>
           
          <span
            class="dashboard-nav__item-icons svelte-1opr8zc"
          >
            
            <svg
              class="dusk-icon dusk-icon--size--default"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M21,9L17,5V8H10V10H17V13M7,11L3,15L7,19V16H14V14H7V11Z"
              />
            </svg>
            
          </span>
           
        </a>
      </nav>
    </div>
     
  </div>
  
   
  <span>
    some text
  </span>
   
  <article
    class="transactions svelte-mwi4ip"
  >
    <header
      class="transactions__header svelte-mwi4ip"
    >
      <h3
        class="h4"
      >
        Transactions
      </h3>
       
      <a
        aria-disabled="false"
        class="dusk-anchor dusk-anchor--on-surface dusk-anchor-button dusk-anchor-button--variant--primary dusk-anchor-button--size--default transactions__footer-button"
        href="/some-base-path/dashboard/transactions"
      >
        <span
          class="dusk-anchor-button__text"
        >
          All transactions
        </span>
        
      </a>
    </header>
     
    <div
      class="dusk-suspense transactions-list__container"
    >
      <dl
        class="transactions-list"
      >
        <dt
          class="transactions-list__term"
        >
          Hash
        </dt>
        <dd
          class="transactions-list__datum transactions-list__datum--hash"
        >
          <samp>
            <a
              class="dusk-anchor dusk-anchor--on-surface"
              href="/explorer/transactions/transaction?id=tg874534tddsghdtst54hdfh795h"
              rel="noopener noreferrer"
              target="_blank"
            >
              tg874534tddsghdtst54hdfh795h
            </a>
          </samp>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Type
        </dt>
        <dd
          class="transactions-list__datum"
        >
          <span
            class="dusk-badge dusk-badge--variant-neutral transactions-list__badge"
          >
            STAKE
          </span>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Block
        </dt>
        <dd
          class="transactions-list__datum"
        >
          33,460
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Amount
        </dt>
        <dd
          class="transactions-list__datum"
        >
          +27,000.00
           
          <span
            class="transactions-list__ticker"
          >
            Dusk
          </span>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Fee
        </dt>
        <dd
          class="transactions-list__datum"
        >
          1,876.00
           
          <span
            class="transactions-list__ticker"
          >
            Dusk
          </span>
           
        </dd>
      </dl>
      <dl
        class="transactions-list"
      >
        <dt
          class="transactions-list__term"
        >
          Hash
        </dt>
        <dd
          class="transactions-list__datum transactions-list__datum--hash"
        >
          <samp>
            <a
              class="dusk-anchor dusk-anchor--on-surface"
              href="/explorer/transactions/transaction?id=tg874534tddsghdtst54hdfh794h"
              rel="noopener noreferrer"
              target="_blank"
            >
              tg874534tddsghdtst54hdfh794h
            </a>
          </samp>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Type
        </dt>
        <dd
          class="transactions-list__datum"
        >
          <span
            class="dusk-badge dusk-badge--variant-neutral transactions-list__badge"
          >
            TRANSFER
          </span>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Block
        </dt>
        <dd
          class="transactions-list__datum"
        >
          33,459
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Amount
        </dt>
        <dd
          class="transactions-list__datum"
        >
          -25,000.00
           
          <span
            class="transactions-list__ticker"
          >
            Dusk
          </span>
           
        </dd>
      </dl>
      <dl
        class="transactions-list"
      >
        <dt
          class="transactions-list__term"
        >
          Hash
        </dt>
        <dd
          class="transactions-list__datum transactions-list__datum--hash"
        >
          <samp>
            <a
              class="dusk-anchor dusk-anchor--on-surface"
              href="/explorer/transactions/transaction?id=tg874534tddsghdtst54hdfh793h"
              rel="noopener noreferrer"
              target="_blank"
            >
              tg874534tddsghdtst54hdfh793h
            </a>
          </samp>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Type
        </dt>
        <dd
          class="transactions-list__datum"
        >
          <span
            class="dusk-badge dusk-badge--variant-neutral transactions-list__badge"
          >
            TRANSFER
          </span>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Block
        </dt>
        <dd
          class="transactions-list__datum"
        >
          33,458
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Amount
        </dt>
        <dd
          class="transactions-list__datum"
        >
          +25,000.00
           
          <span
            class="transactions-list__ticker"
          >
            Dusk
          </span>
           
        </dd>
      </dl>
      <dl
        class="transactions-list"
      >
        <dt
          class="transactions-list__term"
        >
          Hash
        </dt>
        <dd
          class="transactions-list__datum transactions-list__datum--hash"
        >
          <samp>
            <a
              class="dusk-anchor dusk-anchor--on-surface"
              href="/explorer/transactions/transaction?id=tg874534tddsghdtst54hdfh792h"
              rel="noopener noreferrer"
              target="_blank"
            >
              tg874534tddsghdtst54hdfh792h
            </a>
          </samp>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Type
        </dt>
        <dd
          class="transactions-list__datum"
        >
          <span
            class="dusk-badge dusk-badge--variant-neutral transactions-list__badge"
          >
            TRANSFER
          </span>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Block
        </dt>
        <dd
          class="transactions-list__datum"
        >
          33,457
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Amount
        </dt>
        <dd
          class="transactions-list__datum"
        >
          -25,000.00
           
          <span
            class="transactions-list__ticker"
          >
            Dusk
          </span>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Fee
        </dt>
        <dd
          class="transactions-list__datum"
        >
          1,876.00
           
          <span
            class="transactions-list__ticker"
          >
            Dusk
          </span>
           
        </dd>
      </dl>
      <dl
        class="transactions-list"
      >
        <dt
          class="transactions-list__term"
        >
          Hash
        </dt>
        <dd
          class="transactions-list__datum transactions-list__datum--hash"
        >
          <samp>
            <a
              class="dusk-anchor dusk-anchor--on-surface"
              href="/explorer/transactions/transaction?id=tg874534tddsghdtst54hdfh791h"
              rel="noopener noreferrer"
              target="_blank"
            >
              tg874534tddsghdtst54hdfh791h
            </a>
          </samp>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Type
        </dt>
        <dd
          class="transactions-list__datum"
        >
          <span
            class="dusk-badge dusk-badge--variant-neutral transactions-list__badge"
          >
            TRANSFER
          </span>
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Block
        </dt>
        <dd
          class="transactions-list__datum"
        >
          33,456
           
        </dd>
        <dt
          class="transactions-list__term"
        >
          Amount
        </dt>
        <dd
          class="transactions-list__datum"
        >
          -25,000.00
           
          <span
            class="transactions-list__ticker"
          >
            Dusk
          </span>
           
        </dd>
      </dl>
      
      
    </div>
  </article>
  
  
</div>
`;
